### PR TITLE
Add top-level products.meta file

### PR DESCRIPTION
**Pull request by jaym**
The products.meta file describes which products we can deploy, along
with which packages belong to those products. This PR also adds a
package.meta that can be added for any package in its component
directory. With this file, we can describe things about the file like if
it's a data service, what binlinks it should install, etc.

TODO:
- [x] update docs

### CODE REVIEW

File: components/automate-deployment/pkg/services/services.go [hunk 1]
```diff
@@ -3,212 +3,232 @@
...
-					pkg, err := habpkg.FromString(pkgStr)
-					if err != nil {
-						panic(errors.Wrapf(err, "failed to parse hab pkg name '%s' in services list", pkgStr))
-					}
-					pkgs = append(pkgs, pkg)
+func AllServices() ([]habpkg.HabPkg, error) {
+	return serviceList, nil
+}
+
+func ContainsCollection(needle string, haystack []string) bool {
```
Comment (yzl) [inline components/automate-deployment/pkg/services/services.go]: 😹

File: lib/product/parser.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	}
+
+	if err := val.validate(); err != nil {
+		return nil, err
+	}
+
+	return &val, nil
+}
+
+func Parse(repoRootPath string) (*Metadata, error) {
```
Comment (ryancragun) [inline lib/product/parser.go]: Some minimal documentation would be nice in this package

File: .expeditor/verify_private.pipeline.yml [hunk 3]
```diff
@@ -494,7 +494,7 @@ steps:
   - label: "[integration] airgap upgrade"
     command:
       - integration/run_test integration/tests/airgap_upgrade.sh
-    timeout_in_minutes: 20
+    timeout_in_minutes: 30
```
[the remark this answers is not in the document]
Comment (jaym) [reply .expeditor/verify_private.pipeline.yml]: this PR causes a rebuild of both the postgres and ES packages and doing upgrades in that case is sloooow. But given that we end up pinning elsewhere, the rebuild wont affect us when we promote or even get past this pr.

File: components/automate-deployment/pkg/services/services.go [hunk 4]
```diff
@@ -3,212 +3,232 @@
...
 // AllBinds contains all the bindings that the deployment service is aware of
 var AllBinds bind.Binds
 
-// automateFull is the full ServiceCollection object with all the data on
-// packages and services we install/manage
-var automateFull parser.ServiceCollection
+var serviceList []habpkg.HabPkg
+var productList []string
+var packageMetadataMap map[string]*product.Package
+var collectionMap map[string]*product.Collection
```
[the remark this answers is not in the document]
Comment (jaym) [reply components/automate-deployment/pkg/services/services.go]: i swear it was like this when i got here. just building on what the last person did :P

**Pull request merged**